### ISSUE TITLE
Failed to install add-on

### ISSUE THREAD

**Issue by NosSwe** [msp1974/visonic_proxy#29]
The command '/bin/ash -o pipefail -c apk add --no-cache py3-pip=24.0-r2 python3-dev=3.12.7-r0 && apk add --no-cache python3=3.12.7-r0 && rm /usr/lib/python*/EXTERNALLY-MANAGED && python3 -m ensurepip && pip3 install -r /tmp/requirements.txt' returned a non-zero code: 1

**Comment by prowebuk** [comment]
Also getting: The command '/bin/ash -o pipefail -c apk add --no-cache py3-pip=24.0-r2 python3-dev=3.12.7-r0 && apk add --no-cache python3=3.12.7-r0 && rm /usr/lib/python*/EXTERNALLY-MANAGED && python3 -m ensurepip && pip3 install -r /tmp/requirements.txt' returned a non-zero code: 2

Core 2025.1.1
Supervisor 2024.12.3
Operating System 14.1
Frontend 20250106.0

Running odroid n2+

**Comment by prowebuk** [comment]
Hi Mark,

Please don't apologise :-)

I attempted to install the plugin from the repository on the latest home assistant on a dedicated odroid n2+

Not sure if it helps but your script didn't remove /usr/lib/python3.12/EXTERNALLY-MANAGED and /tmp/requirements.txt doesn't exist

Maybe set -o pipefail would help to identify the error?

**Comment by prowebuk** [comment]
This is a Quad-core Cortex-A73
# arch
aarch64

**Comment by msp1974** [comment]
This does build and work on both my dev and prod HA instances, however I am running on amd64 architecture and not aarch64.  I wonder if one of the packages does not have a wheels for aarch64.

I have updated the base image versions to see if that resolves it.  Can you give that a go for me now.  Thx

**Comment by NosSwe** [comment]
installation works fine now for me.
but no connection between panel and HA. Is it normal to show 0.0.0.0 for the listeners?

`2025-01-08 15:18:28,282 INFO     Setting visonic_host to connect.tycomonitor.com from config
2025-01-08 15:18:28,283 INFO     Setting LOG_LEVEL to info from config
2025-01-08 15:18:28,283 INFO     Setting websocket_mode to True from config
2025-01-08 15:18:28,283 INFO     Setting log_level to info from config
2025-01-08 15:18:28,283 INFO     Setting message_log_level to 1 from config
2025-01-08 15:18:28,283 INFO     Proxy Mode: True
2025-01-08 15:18:28,283 INFO     Websocket Mode: True
2025-01-08 15:18:28,283 INFO     Log Level: INFO
2025-01-08 15:18:28,283 INFO     Message Log Level: 3
2025-01-08 15:18:28,283 INFO     Proxy Server is starting
2025-01-08 15:18:28,283 INFO     Starting Connection Manager
2025-01-08 15:18:28,283 INFO     Command Manager started
2025-01-08 15:18:28,283 INFO     Message Router started
2025-01-08 15:18:28,283 INFO     Flow Manager started
2025-01-08 15:18:28,283 INFO     Listening for Alarm connection on 0.0.0.0 port 5001
2025-01-08 15:18:28,283 INFO     Started Alarm Watchdog Timer
2025-01-08 15:18:28,284 INFO     Websocket server listening on 0.0.0.0 port 8082
2025-01-08 15:18:28,284 INFO     Starting Webserver
2025-01-08 15:18:28,284 INFO     Webserver listening on 0.0.0.0 port 8443
2025-01-08 15:18:28,284 INFO     Connection Manager started
2025-01-08 15:18:28,284 INFO     Proxy Server running`

**Comment by prowebuk** [comment]
Installs fine for me too thanks Mark, will configure when I get home and let you know how I get on.

**Comment by msp1974** [comment]
@NosSwe, yes that is normal as 0.0.0.0 means listening on all IPs for that container.  Have you configured your panel and forced it to disconnect from the PowerManage servers?

**Comment by NosSwe** [comment]
ok good.
I'm actually testing with a DSC PowerSeries Neo panel, which I normally have receiver set to "connect.tycomonitor.com".
I've set the addon config "visonic_host:" to "connect.tycomonitor.com" and the panel is set to my HA IP on both receiver 1 and 2 during testing.

**Comment by msp1974** [comment]
Ah ok.  I just tried that dns name on mine and it did connect on port 5001 fine but is obvioulsy a different service to the one I use (which is the default IP address in the visonic host).

Sorry have no experience of this panel.  Are you able to see through your router/app if this is no longer connecting to the internet service?  Or if it is, is it connecting on port 5001 & 8443?

**Comment by NosSwe** [comment]
I now tried setting receiver1 back to connect.tycomonitor.com (leaving receiver2 set to HA ip) and blocking panel internet access with a traffic rule. resulting in a notification for blocked traffic with destination port 1303 every time. I tried changing the add-on config "alarm connection" port from 5001 to 1303 -but the add-on start up log output still prints "Listening for Alarm connection on 0.0.0.0 port 5001"

**Comment by msp1974** [comment]
Yes, it will show listening on port 5001 but the container should be exposing port 1303 redirected to 5001 so allowing connection from your panel to 1303.  Let me check that

**Comment by msp1974** [comment]
Looking at this,

https://www.dsc.com/connect-alarm/media/docs/Connect-Alarm-Panel-Set-Up.pdf

it seems there are 2 different ports - 3061 as remote port and 1303 as Real time notification port.

When looking at the Visonic PowerManage server documentation, these ports are not mentioned.  So, I wonder if this is a different server @ tycomonitor your alarm connects to but (if I remember correctly, the rest api integration works for you) has the same rest api interface.

So, I am wondering if this works very differently to the Powerlink3.1 module?  You could try running it as a local docker and amend some of the ports to see if that works (all ports are configured in const.py)  as in the addon, it wont connect to tycomonitor on anything but 5001, but I suspect this may not be compatible.

The messages it is reading and writing are also very much Visonic proprietary messages I think.

**Comment by prowebuk** [comment]
Hi Dave,

Had to replace my PowerMaster 33 panel due to old firmware which wasn't compatible with PowerLink 3.1 and this was cheaper than getting somebody out to upgrade it :-)
Now your proxy is working perfectly with HA and so is the Visonic Go and Visonic Alarm Install app (took me a while to realise you could register for an installer account allowing remote panel config when access is enabled in Visonic Go app) so many thanks for a great solution that just works.

Kind regards

Richard

**Comment by msp1974** [comment]
I'm going to close this issue as install on aarch64 is now working.

@NosSwe, after doing more reading, your DSC Neo is not compatible with this addon and never will be as it uses a completely different protocol with encryption, from research I have done.  Sorry about that.
